File: Backend/Events/Converters/Parties/ResistancesConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Party.Digimon;

namespace Backend.Events.Converters.Parties;

public static class ResistancesConverter
{
    public static ResistancesDTO ToDTO(Resistances resistances) => new()
    {
        Fire = resistances.Fire,
        Water = resistances.Water,
        Ice = resistances.Ice,
        Wind = resistances.Wind,
        Thunder = resistances.Thunder,
        Machine = resistances.Machine,
        Dark = resistances.Dark
    };
}

File: Frontend/src/events/converters/parties/digimons/resistances.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { ResistancesDTO } from "@/events/dto/parties/digimons/resistances.dto";
import type { Resistances } from "@/models";

export class ResistancesConverter {
  public static convert(resistancesDto: DeepRequired<ResistancesDTO>): Resistances {
    return {
      fire: resistancesDto.fire,
      water: resistancesDto.water,
      ice: resistancesDto.ice,
      wind: resistancesDto.wind,
      thunder: resistancesDto.thunder,
      machine: resistancesDto.machine,
      dark: resistancesDto.dark,
    };
  }
}
